packages/opentelemetry-sdk-setup/src/util.ts

```diff
@@ -0,0 +1,122 @@
+  // logs.
+  const fileDescriptor = createWriteStream('', { fd: systemLogFile })
+
+  fileDescriptor.on('error', () => {
+    console.error('Could not write to system log file')
+  })
+
+  return (...args) => fileDescriptor.write(`${console.log(...args)}\n`)
+}
+
+/** Given a simple logging function return a `DiagLogger`. Used to setup our system logger as the diag logger.*/
+export const getOtelLogger = function (
+  debug: boolean,
+  /** A system log file descriptor, if non is provided it will be a noop logger */
+  systemLogFile?: number,
+): DiagLogger {
+  const logger = getSystemLogger(debug, systemLogFile)
+  const otelLogger = (...args: any[]) => {
+    // Debug log msgs can be an array of 1 or 2 elements with the second element being an array fo multiple elements
+    const msgs = args.flat(1)
+    logger('[otel-traces]', ...msgs)
+  }
+  return {
+    debug: otelLogger,
+    info: otelLogger,
+    error: otelLogger,
+    verbose: otelLogger,
+    warn: otelLogger,
+  }
+}
+
+//** Loads the baggage attributes from a baggabe file which follows W3C Baggage specification */
+export const loadBaggageFromFile = async function (baggageFilePath?: string) {
+  if (baggageFilePath === undefined || baggageFilePath.length === 0) {
+    diag.warn('No baggage file path provided, no context loaded')
+    return {}
+  }
+  let baggageString: string
+  try {
+    baggageString = await readFile(baggageFilePath, 'utf-8')
+  } catch (error) {
+    diag.error(error)
+    return {}
+  }
+  return parseKeyPairsIntoRecord(baggageString)
+}
+
+/**
+ * Given a path to a node executable (potentially a symlink) get the module packageJson
+ */
+export const findExecutablePackageJSON = async function (path: string): Promise<PackageJson> {
+  let pathToSearch: string
+  try {
+    // resolve symlinks
+    pathToSearch = await realpath(path)
+  } catch {
+    // bail early if we can't resolve the path
+    return {}
+  }
+
+  try {
+    const result = await readPackageUp({ cwd: pathToSearch, normalize: false })
+    if (result === undefined) return {}
+    const { packageJson } = result
+    return packageJson
+  } catch {
+    // packageJson read failed, we ignore the error and return an empty obj
+    return {}
+  }
+}
+
+/**
+ * Sets global context to be used when initialising our root span
+ * TODO this will move to a shared package (opentelemetry-utils) to scope the usage of this global property there
+ */
+export const setGlobalContext = function (ctx: Context) {
+  global['NETLIFY_GLOBAL_CONTEXT'] = ctx
+}
+
+/**
+ * Gets the global context to be used when initialising our root span
+ * TODO this will move to a shared package (opentelemetry-utils) to scope the usage of this global property there
+ */
+export const getGlobalContext = function (): Context {
+  if (global['NETLIFY_GLOBAL_CONTEXT'] === undefined) {
+    return context.active()
+  }
+  return global['NETLIFY_GLOBAL_CONTEXT']
+}
```

Comment:
These methods will also be moved to the opentelemetry utility package, as way to scope as much as we can the `global` use (all the interactions should go through these methods).

Once we do the initial rollout of this module, we'll work on moving the remainder of the pieces of the methods in build (and this global context interaction methods) to a `opentelemetry-util` module, which only depends on `@opentelemetry/api` and both modules (build and this) can depend on.